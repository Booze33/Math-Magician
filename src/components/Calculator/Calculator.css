body {
  margin: 0;
  display: flex;
  padding-top: 10%;
  justify-content: center;
  align-items: center;
}

.calculator {
  width: 35vw;
  height: 24rem;
  background-color: #000;
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: repeat(6, 1fr);
}

.div {
  height: 4rem;
  width: 100%;
  background-color: pink;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Display {
  background-color: #787889;
  display: flex;
  justify-content: end;
  color: #fff;
}

.grid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: 100%;
  background-color: #dddddf;
}

.number {
  height: 100%;
}

.color {
  background-color: orange;
}

.number-0 {
  grid-column: span 2;
}
