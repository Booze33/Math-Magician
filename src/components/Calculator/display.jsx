import React from 'react';
import PropTypes from 'prop-types';

function Display({ value }) {
  return (
    <div className="Display div">{value}</div>
  );
}

Display.propTypes = {
  value: PropTypes.string.isRequired,
};

export default Display;
