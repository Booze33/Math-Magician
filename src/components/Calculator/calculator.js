import React, { useState } from 'react';
import Display from './display';
import calculate from './logic/calculate';
import './Calculator.css';

function Calculator() {
  const [calculator, setCalculator] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setCalculator(calculate(calculator, e.target.innerText));
  };

  const displayValue = () => {
    if (calculator.next) {
      return Number(calculator.next).toLocaleString();
    }
    if (calculator.total) {
      return Number(calculator.total).toLocaleString();
    }

    return '0';
  };

  return (
    <div className="calculator">
      <Display value={displayValue()} />
      <div className="frs-div div grid">
        <button className="number number-AC" type="button" onClick={handleClick}>AC</button>
        <button className="number number-+-" type="button" onClick={handleClick}>+/-</button>
        <button className="number number-%" type="button" onClick={handleClick}>%</button>
        <button className="number number-// color" type="button" onClick={handleClick}>รท</button>
      </div>
      <div className="sec-div div grid">
        <button className="number number-7" type="button" onClick={handleClick}>7</button>
        <button className="number number-8" type="button" onClick={handleClick}>8</button>
        <button className="number number-9" type="button" onClick={handleClick}>9</button>
        <button className="number number-x color" type="button" onClick={handleClick}>x</button>
      </div>
      <div className="trd-div div grid">
        <button className="number number-4" type="button" onClick={handleClick}>4</button>
        <button className="number number-5" type="button" onClick={handleClick}>5</button>
        <button className="number number-6" type="button" onClick={handleClick}>6</button>
        <button className="number number-- color" type="button" onClick={handleClick}>-</button>
      </div>
      <div className="frt-div div grid">
        <button className="number number-1" type="button" onClick={handleClick}>1</button>
        <button className="number number-2" type="button" onClick={handleClick}>2</button>
        <button className="number number-3" type="button" onClick={handleClick}>3</button>
        <button className="number number-+ color" type="button" onClick={handleClick}>+</button>
      </div>
      <div className="fth-div div grid">
        <button className="number number-0" type="button" onClick={handleClick}>0</button>
        <button className="number number---" type="button" onClick={handleClick}>.</button>
        <button className="number number-num color" type="button" onClick={handleClick}>=</button>
      </div>
    </div>
  );
}

export default Calculator;
