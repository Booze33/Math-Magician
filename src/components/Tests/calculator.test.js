import renderer from 'react-test-renderer';
import Calculator from '../Calculator/calculator';

test('Test paragraph-1', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
