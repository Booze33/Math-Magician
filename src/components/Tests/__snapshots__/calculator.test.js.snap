// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test paragraph-1 1`] = `
<div
  className="calculator"
>
  <div
    className="Display div"
  >
    0
  </div>
  <div
    className="frs-div div grid"
  >
    <button
      className="number number-AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="number number-+-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="number number-%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="number number-// color"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="sec-div div grid"
  >
    <button
      className="number number-7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="number number-8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="number number-9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="number number-x color"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="trd-div div grid"
  >
    <button
      className="number number-4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="number number-5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="number number-6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="number number-- color"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="frt-div div grid"
  >
    <button
      className="number number-1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="number number-2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="number number-3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="number number-+ color"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="fth-div div grid"
  >
    <button
      className="number number-0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="number number---"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="number number-num color"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
